### ISSUE TITLE
Enable Link-Time Optimization (LTO) for the Tauri part

### ISSUE THREAD

**Issue by zamazan4ik** [brutusyhy/polars-explorer#15]
Hi!

I noticed that in the `Cargo.toml` [file](https://github.com/brutusyhy/polars-explorer/blob/main/polars-explorer/src-tauri/Cargo.toml) Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where additionally to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO should also help to reduce the build-time overhead with LTO. E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

According to the official Tauri [documentation](https://v1.tauri.app/v1/guides/building/app-size/#rust-build-time-optimizations), you may also be interested in enabling other optimizations.

Thank you.

**Comment by brutusyhy** [comment]
Thank you for the suggestion! I haven't considered this part as I think there's still a long way to go before the app can reach a release status, but I will follow your advice!

**Comment by brutusyhy** [comment]
By the way, do you have suggestions on how to improve compilation speed? This has been a huge limiting factor for me, as even minor changes to the backend code would require 5-10 minutes of recompilation. I have tried a few configurations, but their effects seem minor

**Comment by zamazan4ik** [comment]
Firstly, I would recommend starting with profiling the building process with https://doc.rust-lang.org/cargo/reference/timings.html . I can guess several options that could help with compilation speed:

* If your problem is the build time during the development process, I hope you are already using the dev profile that doesn't enable expensive from the build time perspective optimizations
* If you are talking about CI build times, configuring proper caching could help you.
* You could try to enable Cranelift backed for the Dev profile - https://github.com/rust-lang/rustc_codegen_cranelift It works faster than the LLVM backend but has fewer optimizations. It could be a good tradeoff for you
* In extreme cases, maybe distributed builds could be a good option for you but I don't have enough experience with them in the Rust ecosystem
